compiler/rustc_middle/src/mir/pretty.rs

```diff
@@ -149,37 +148,58 @@ pub fn dump_enabled(tcx: TyCtxt<'_>, pass_name: &str, def_id: DefId) -> bool {
 // `def_path_str()` would otherwise trigger `type_of`, and this can
 // run while we are already attempting to evaluate `type_of`.
 
+/// Most use-cases of dumping MIR should use the [dump_mir] entrypoint instead, which will also
+/// check if dumping MIR is enabled, and if this body matches the filters passed on the CLI.
+///
+/// That being said, if the above requirements have been validated already, this function is where
+/// most of the MIR dumping occurs, if one needs to export it to a file they have created with
+/// [create_dump_file], rather than to a new file created as part of [dump_mir].
+pub fn dump_mir_to_writer<'tcx, F>(
```

Comment:
I've wanted this so I could dump MIR to stderr :)

Reply:
Darn I’ve also wanted that before, I’ll actually add it to the comment.

Reply:
Done